oci/spec_opts.go

```diff
@@ -454,13 +454,12 @@ func WithUserNamespace(container, host, size uint32) SpecOpts {
 				Type: specs.UserNamespace,
 			})
 		}
-		mapping := specs.LinuxIDMapping{
-			ContainerID: container,
-			HostID:      host,
-			Size:        size,
-		}
-		s.Linux.UIDMappings = append(s.Linux.UIDMappings, mapping)
-		s.Linux.GIDMappings = append(s.Linux.GIDMappings, mapping)
+		uidMapCopy := make([]specs.LinuxIDMapping, len(uidMap))
+		gidMapCopy := make([]specs.LinuxIDMapping, len(gidMap))
+		copy(uidMapCopy, uidMap)
```

Comment:
why copy?

Reply:
I think that if users passed in arrays of the `specs.LinuxIDMapping` to `WithUserNamespace` and then change the values in the array after the call, the values will not be the same as what was passed in to `WithUserNamespace`.

Reply:
I removed it instead. I don't think users will do that.